### PR TITLE
Apply f-string formatting to everything (except logging)

### PR DESCRIPTION
**Pull request by ShadowJonathan**
## What was wrong?

Issue #382 

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: libp2p/transport/tcp/tcp.py [hunk 1]
```diff
@@ -91,5 +100,52 @@ def create_listener(self, handler_function: THandler) -> TCPListener:
...
+        # ipv6: (host, port, flowinfo, scopeid)
+        addr, port = sock.getsockname()[:2]
+        ip = p_code(P_IP6).name
+
+    else:
+        raise NotImplementedError(
+            f"Cannot convert socket to multiaddr, socket family is of {sock.family}"
+        )
+
+    return Multiaddr(f"/{ip}/{addr}/{t_proto}/{port}")
```
Comment (ShadowJonathan) [inline libp2p/transport/tcp/tcp.py]: I started changing `Multiaddr("/ip4/%s/tcp/%s" % socket.getsockname())` till I realized it needed an actual working function, so I made one.

Comment (pipermerriam) [reply libp2p/transport/tcp/tcp.py]: Typically we would request that this type of substantive change to functionality be submitted separately from the other changes.  Do you think you could extract this refactor into it's own PR that we'll review and merge separately and this branch can be based on top of that other branch?

Comment (ShadowJonathan) [reply libp2p/transport/tcp/tcp.py]: Yes, I'll do that.

File: libp2p/crypto/ecc.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def get_type(self) -> KeyType:
         return KeyType.ECC_P256
 
     def verify(self, data: bytes, signature: bytes) -> bool:
-        raise NotImplementedError
+        raise NotImplementedError()
```
Comment (mhchia) [inline libp2p/crypto/ecc.py]: Actually `raise NotImplementedError` works: https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

File: libp2p/tools/interop/daemon.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def __init__(
         is_pubsub_signing: bool = False,
         is_pubsub_signing_strict: bool = False,
     ) -> None:
-        args = [f"-listen={str(control_maddr)}"]
+        args = [f"-listen={control_maddr!s}"]
```
Comment (mhchia) [inline libp2p/tools/interop/daemon.py]: Note: IMO `str(control_maddr)` is more explicit, but never mind.

Comment (ShadowJonathan) [reply libp2p/tools/interop/daemon.py]: More bytes. 

Also `!s` is proper for brace formatting over `str`, and they're essentially the same thing internally.

File: libp2p/crypto/ecc.py (hunk 2 above)
Comment (ShadowJonathan) [reply libp2p/crypto/ecc.py]: Huh, thanks for telling. I personally like putting braces after them (to be more explicit), but I'll reverse these changes if that's requested.

**Comment by mhchia** [comment]
Would you mind fixing the conflict so that I can merge it? Thanks!

**Comment by ShadowJonathan** [comment]
> Would you mind fixing the conflict so that I can merge it? Thanks!
Fixed 👍

**Comment by mhchia** [comment]
Sorry, that I failed to merge it previously. Would you mind fixing the conflict again? 😢 Thank you!

**Comment by ShadowJonathan** [comment]
@mhchia fixed it 👍

**Comment by mhchia** [comment]
Thank you! Please ignore the failure of `tests_interop` for now, due to some breaking changes in the dependency `p2pclient`.

**Pull request merged**